src/Bicep.LangServer/Handlers/BicepDeploymentScopeRequestHandler.cs

```diff
@@ -134,6 +134,8 @@ private Compilation GetCompilation(DocumentUri documentUri)
         {
             var fileUri = documentUri.ToUri();
 
+            // Bicep file could contain load functions like loadTextContent(..). We'll refresh compilation to detect changes in files referenced in load functions.
```

Comment:
@anthony-c-martin @majastrz Could you please review this PR?  I assume the changes are probably sufficient to fix deployment from vscode.  But is this indicative of a deeper problem that should be fixed elsewhere?  E.g. should we be watching for changes to loadTextContent()/etc targets?  Not sure if it affects anything else.

@bhsubra Does this not also affect the build command?

Reply:
`RefreshCompilation()` was added so we could recompile after module restore is performed because there isn't a small set of files to watch in that case.

We're able to recompile as needed when files that the module depends on change today. The existing logic handles bicep files and ARM template files, but not anything that load...() functions load. It would be good to fix that problem this way and not rely on `RefreshCompilation()`

The tricky part will be that other file linking is done via syntax before the semantic model is created, but this logic would have to happen after binding because you need symbols to identify load...() functions.

I am also curious why we didn't have to do it for the build command.

Reply:
Btw, @stan-sz is looking to solve a similar problem on [https://github.com/Azure/bicep/issues/5715](https://github.com/Azure/bicep/issues/5715), so it'd be good to collaborate on a shared fix.

Reply:
Should bicep have a new command to list all inputs?

Reply:
Had an offline discussion with @puicchan and @StephenWeatherford and created https://github.com/Azure/bicep/issues/8785 for addressing the root cause. @majastrz, @anthony-c-martin , are you guys ok with taking in this fix for now?

Reply:
Yup, I'm totally fine with this fix!

Reply:
+1 I'm also fine with the stopgap solution.

Reply:
> Should bicep have a new command to list all inputs?

Yes, I think we should.

Reply:
Thanks @anthony-c-martin and @majastrz

Reply:
@stan-sz, I created https://github.com/Azure/bicep/issues/8803 to add a new command to list all inputs. Thank you!